components/AddDonation.js

```diff
@@ -0,0 +1,136 @@
+import styled from "styled-components";
+import { useState } from "react";
+import Counter from "./Counter";
+
+export default function AddDonation({ createDonation }) {
+  const [toggleAddButton, setToggleAddButton] = useState(true);
+  const [count, setCount] = useState(0);
+  function handleClick() {
+    setToggleAddButton(!toggleAddButton);
+  }
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    createDonation(data);
+    setCount(0);
+    event.target.reset();
+    setToggleAddButton(!toggleAddButton);
+  }
+
+  return (
+    <StyledDiv>
+      {toggleAddButton ? (
+        <StyledButton onClick={handleClick}>
+          Neue Spende hinzufügen
+        </StyledButton>
+      ) : (
+        <StyledForm onSubmit={handleSubmit}>
+          <label htmlFor="category">Kategorie:</label>
```

Comment:
When you adapt the boolean as described above you need to change either the order of button and form or invert the boolean here, e.g.: `{ !isFormVisible ? ( 
  button ...
) : (
form ...
)}`

Reply:
👍🏽